config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -324,19 +324,32 @@ private void loadFolderDependencies(AnnotatedBundle annotatedBundle, GatewayEnti
      */
     private void loadGatewayEntity(Dependency dependency, AnnotatedBundle annotatedBundle, Bundle rawBundle) {
         Class<? extends GatewayEntity> entityClass = entityTypeRegistry.getEntityClass(dependency.getType());
-        Map<String, ? extends GatewayEntity> allEntitiesOfType = rawBundle.getEntities(entityClass);
-        Optional<? extends Map.Entry<String, ? extends GatewayEntity>> optionalGatewayEntity =
-                allEntitiesOfType.entrySet().stream()
-                        .filter(e -> {
-                            GatewayEntity gatewayEntity = e.getValue();
-                            if (gatewayEntity.getName() != null) {
-                                return dependency.getName().equals(gatewayEntity.getName());
-                            } else {
-                                return dependency.getName().equals(PathUtils.extractName(e.getKey()));
-                            }
-                        }).findFirst();
-        Map entityMap = annotatedBundle.getEntities(entityClass);
-        optionalGatewayEntity.ifPresent(e -> entityMap.put(e.getKey(), e.getValue()));
+        if(entityClass != null){
+            Map<String, ? extends GatewayEntity> allEntitiesOfType = rawBundle.getEntities(entityClass);
```

Comment:
@uppoju , can you please change the rawBundle name to fullBundle in this PR, it is actually confusing with actual rawBundle which we get at the time of export.

Reply:
We dont get raw bundle out of export, its a resultant bundle